src/backend/distributed/commands/multi_copy.c

```diff
@@ -845,7 +1055,8 @@ OpenCopyTransactions(CopyStmt *copyStatement, ShardConnections *shardConnections
 	/* otherwise, mark failed placements as inactive: they're stale */
 	foreach(failedPlacementCell, failedPlacementList)
 	{
-		ShardPlacement *failedPlacement = (ShardPlacement *) lfirst(failedPlacementCell);
+		ShardPlacement *failedPlacement = (ShardPlacement *) lfirst(
+			failedPlacementCell);
 		uint64 shardLength = 0;
 
 		DeleteShardPlacementRow(failedPlacement->shardId, failedPlacement->nodeName,
```

Comment:
> Seems like we shouldn't do this for COPY via worker.

I modified this function to get a stopOnFailure parameter and for COPY via worker code never hits this block. Should I add a comment to here or move stopOnFailure check to here? I slighly inclined for a comment.

		if (connection == NULL)
		{
			if (stopOnFailure)
			{
				ereport(ERROR, (errmsg("could not open connection to %s:%d",
									   nodeName, nodePort)));
			}

			failedPlacementList = lappend(failedPlacementList, placement);
			continue;
		}